pkg/provider/gitlab/acl_test.go

```diff
@@ -130,3 +132,107 @@ func TestIsAllowed(t *testing.T) {
 		})
 	}
 }
+
+func TestMembershipCaching(t *testing.T) {
+	ctx, _ := rtesting.SetupFakeContext(t)
+
+	v := &Provider{
+		targetProjectID: 3030,
+		userID:          4242,
+	}
+
+	client, mux, tearDown := thelp.Setup(t)
+	defer tearDown()
+	v.gitlabClient = client
+
+	// Count how many times the membership API is hit.
+	var calls int
+	thelp.MuxAllowUserIDCounting(mux, v.targetProjectID, v.userID, &calls)
+
+	ev := &info.Event{Sender: "someone", PullRequestNumber: 1}
+
+	// First call should hit the API once and cache the result.
+	allowed, err := v.IsAllowed(ctx, ev)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if !allowed {
+		t.Fatalf("expected allowed on first membership check")
+	}
+	if calls < 1 {
+		t.Fatalf("expected at least 1 membership API call, got %d", calls)
+	}
+
+	// Second call should use the cache and not hit the API again.
+	allowed, err = v.IsAllowed(ctx, ev)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if !allowed {
+		t.Fatalf("expected allowed on cached membership check")
+	}
+	if calls != 1 {
+		t.Fatalf("expected cached result with no extra API call, got %d calls", calls)
+	}
+}
+
+func TestMembershipAPIFailureDoesNotCacheFalse(t *testing.T) {
```

Comment:
Test name is slightly misleading; if the API successfully returns false, it will/should be cached (assuming the owners file check also returns false). 
```suggestion
func TestMembershipAPIFailureDoesNotCacheApiError(t *testing.T) {
```